packages/cozy-harvest-lib/src/services/biWebView.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Interface between Budget Insight and Cozy using BI's webview
+ *
+ * - Deals with the konnector to get temporary tokens
+ */
+
+import get from 'lodash/get'
+import clone from 'lodash/clone'
+
+import { getBIConnection } from './bi-http'
+import assert from '../assert'
+import logger from '../logger'
+import flag from 'cozy-flags'
+import {
+  fetchExtraOAuthUrlParams,
+  createTemporaryToken,
+  setBIConnectionId,
+  saveBIConfig,
+  findAccountWithBiConnection,
+  convertBIErrortoKonnectorJobError
+} from './budget-insight'
+import { KonnectorJobError } from '../helpers/konnectors'
+
+export const isBiWebViewConnector = konnector =>
+  flag('harvest.bi.webview') &&
+  get(konnector, 'partnership.domain') === 'budget-insight.com'
+
+/**
+ * Runs multiple checks on the bi connection referenced in the given account
+ *
+ * @param {Object} options.account The account content
+ * @param {Object} options.flow
+ * @param {Object} options.konnector connector manifest content
+ * @param {Object} options.client CozyClient object
+ *
+ * @return {Integer} Connection Id
+ */
+export const checkBIConnection = async ({
+  account,
+  client,
+  konnector,
+  flow
+}) => {
+  try {
+    let connId = getWebviewBIConnectionId(account)
+
+    logger.info('Creating temporary token...')
+
+    const biConfig = await createTemporaryToken({
+      client,
+      konnector,
+      account
+    })
+    saveBIConfig(flow, biConfig)
+
+    const { code: tempToken, ...config } = biConfig
+
+    logger.info('Created temporary token')
+    assert(tempToken, 'No temporary token')
+
+    logger.info(`fetch connection ${connId}...`)
+
+    const connection = await getBIConnection(config, connId, tempToken)
+
+    const sameAccount = await findAccountWithBiConnection({
+      client,
+      konnector,
+      connectionId: connection.id
+    })
+    if (sameAccount) {
+      const err = new KonnectorJobError(
+        'ACCOUNT_WITH_SAME_IDENTIFIER_ALREADY_DEFINED'
+      )
+      err.accountId = sameAccount._id
+      throw err
+    }
+    return connection
+  } catch (err) {
+    return convertBIErrortoKonnectorJobError(err)
+  }
+}
+
+/**
+ * Handles webview connection
+ *
+ * @param {Object} options.account The account content
+ * @param {Object} options.flow
+ * @param {Object} options.konnector connector manifest content
+ * @param {Object} options.client CozyClient object
+ *
+ * @return {Integer} Connection Id
+ */
+export const handleOAuthAccount = async ({
+  account,
+  flow,
+  konnector,
+  client,
+  t
+}) => {
+  let biWebviewAccount = clone(account)
+  const cozyBankId = getCozyBankId({ konnector, account })
+  if (cozyBankId) {
+    set(biWebviewAccount, 'auth.bankId', cozyBankId)
+  }
+
+  const connectionId = getWebviewBIConnectionId(biWebviewAccount)
+
+  if (connectionId) {
+    logger.info(`Found a bi webview connection id: ${connectionId}`)
```

Comment:
```suggestion
    logger.info(`Found a BI webview connection id: ${connectionId}`)
```

in order to make the log clear for log reader that is not familar with BI